### PR TITLE
Fix Compatibility Issue: check for vscode-spring-boot API

### PR DESCRIPTION
**Pull request by vudayani-vmw**

### CODE REVIEW

**Comment by vudayani-vmw** [comment]
@Eskibear 
This is in reference to #244
I tested with vscode-spring-boot (v1.39.0, v1.38.0) which didn’t have the metrics API and the memory view doesn't show up.
Is there a particular version where it is failing?

Currently, the check to show the memory view happens in [updateProcessMemoryMetrics](https://github.com/vudayani-vmw/vscode-spring-boot-dashboard/blob/afd0ea639ddcb3a2391d93c8151c1fb207b530c0/src/controllers/LiveDataController.ts#L70) when the metric results are published for the live process. The memory view should not show up if the metrics API is missing.
I have also added the below [check](https://github.com/vudayani-vmw/vscode-spring-boot-dashboard/blob/afd0ea639ddcb3a2391d93c8151c1fb207b530c0/src/models/stsApi.ts#L67) on the API call. Will this check suffice?
```
if(typeof stsApi.getLiveProcessMetricsData === "function") {
        return await stsApi.getLiveProcessMetricsData({
            processKey: processKey,
            endpoint: "metrics",
            metricName: "gcPauses"
        });
    }
```

File: src/controllers/LiveDataController.ts [hunk 1]
```diff
@@ -65,10 +73,13 @@ async function updateProcessMemoryMetrics(payload: string | LocalLiveProcess) {
 
     const heapMemMetrics = await getMemoryMetrics(processKey, "heapMemory");
     const nonHeapMemMetrics = await getMemoryMetrics(processKey, "nonHeapMemory");
-    await vscode.commands.executeCommand("setContext", "spring.memoryGraphs:hasLiveProcess", liveProcess !== undefined);
-    memoryProvider.refreshLiveHeapMemoryMetrics(liveProcess, heapMemMetrics);
-    memoryProvider.refreshLiveNonHeapMemoryMetrics(liveProcess, nonHeapMemMetrics);
-    store.data.set(processKey, { heapMemMetrics, nonHeapMemMetrics});
+
+    if(heapMemMetrics || nonHeapMemMetrics) {
+        await vscode.commands.executeCommand("setContext", "spring.memoryGraphs:hasLiveProcess", liveProcess !== undefined);
```
Comment (Eskibear) [inline src/controllers/LiveDataController.ts:78]: I think this check is what I expect. Previously it just shows the memory view without plotting any data points.

File: src/controllers/LiveDataController.ts [hunk 2]
```diff
@@ -24,8 +24,14 @@ export async function init() {
     stsApi.onDidLiveProcessDisconnect(resetProcessInfo);
     stsApi.onDidLiveProcessUpdate(updateProcessInfo);
     // proposed API on vscode-spring-boot extension. check before calling.
-    stsApi.onDidLiveProcessGcPausesMetricsUpdate?.(updateProcessGcPausesMetrics);
-    stsApi.onDidLiveProcessMemoryMetricsUpdate?.(updateProcessMemoryMetrics);
+    if(typeof stsApi.onDidLiveProcessGcPausesMetricsUpdate === "function") {
```
Comment (Eskibear) [inline src/controllers/LiveDataController.ts:27]: previous statement is sufficient enough for me, you make it safer by checking whether the api is a function.

File: src/controllers/LiveDataController.ts [hunk 3]
```diff
@@ -24,8 +24,14 @@ export async function init() {
...
+        stsApi.onDidLiveProcessGcPausesMetricsUpdate?.(updateProcessGcPausesMetrics);
```
Comment (Eskibear) [inline src/controllers/LiveDataController.ts:28]: with above check, here you don't need `?`, which is a null check.

btw `func?.()` roughly equals to `func ? func() : func`

Comment (vudayani-vmw) [reply src/controllers/LiveDataController.ts]: I have reverted the check to use `stsApi.onDidLiveProcessGcPausesMetricsUpdate?.(updateProcessGcPausesMetrics);`

**Comment by vudayani-vmw** [comment]
@Eskibear 
I have done a round of testing with the previous as well as the current stable versions and the memory view seems to work as expected. Please let me know if there are any further changes to be made :-)

**Comment by Eskibear** [comment]
Thanks. I'm merging it, so you'll see it in next pre-release.

**Pull request merged**